### PR TITLE
datajoint/table.py: smarter dataframe conversion (#666)

### PR DESCRIPTION
**Pull request by ixcat**
to address #666

needs further discussion/confirmation with those more familiar with pandas

### CODE REVIEW

**Comment by ixcat** [comment]
@dimitri-yatsenko / @eywalker / @guzman-raphael -
Would be good to get some feedback here before un-WIP, my familiarity with pandas is a bit lacking. See also example test script `issue666.txt` in #666 comments.

**Comment by ixcat** [comment]
Thanks for the feedback @gucky92 
Just to make sure I've got it - these both refer to passing the 'drop' argument to reset_index, correct? e.g. you're proposing either:

* `rows.reset_index(drop=not isinstance(rows.index, pandas.MultiIndex)).to_records(index=False)`
or:
* `rows.reset_index(drop=set(self.primary_key) != set(rows.index.names)).to_records(index=False)`

correct?

**Comment by gucky92** [comment]
@ixcat Yes, that is what i meant.

**Comment by eywalker** [comment]
Simpler and more in alignment with what we actually want to check is whether there is any "named" index/indices. Given this, a good check would be 
```
rows.reset_index(drop=len(rows.index.names)==1 and not rows.index.names[0]).to_records(index=False)
```
The default case of an unnamed index will have `rows.index.names` with single entry of `None` and can be safely dropped.

File: datajoint/table.py [hunk 1]
```diff
@@ -193,7 +193,9 @@ def insert(self, rows, replace=False, skip_duplicates=False, ignore_extra_fields
         """
 
         if isinstance(rows, pandas.DataFrame):
-            rows = rows.to_records()
+            rows = rows.reset_index(
```
Comment (eywalker) [inline datajoint/table.py:196]: Suggestion is to check for actual empty named index to drop:
```python
rows.reset_index(drop=len(rows.index.names)==1 and not rows.index.names[0]).to_records(index=False)
```

**Comment by ixcat** [comment]
back to 'WIP' until unit test

**Comment by eywalker** [comment]
Ah sorry I removed WIP as this was set as Draft and figured that suffice to indicate the WIP status.

**Pull request merged**